From 81a23859ef6de8765e57fc047e103f98126c26d1 Mon Sep 17 00:00:00 2001
From: Ondrej Dubaj <odubaj@redhat.com>
Date: Tue, 16 Apr 2019 11:17:01 +0200
Subject: [PATCH] fixed insecure buffer copy

---
 odbcinst/SQLWriteFileDSN.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/odbcinst/SQLWriteFileDSN.c b/odbcinst/SQLWriteFileDSN.c
index c22a4bf..9891c73 100644
--- a/odbcinst/SQLWriteFileDSN.c
+++ b/odbcinst/SQLWriteFileDSN.c
@@ -21,7 +21,7 @@ BOOL SQLWriteFileDSN(			LPCSTR	pszFileName,
 
 	if ( pszFileName[0] == '/' )
 	{
-		strcpy( szFileName, pszFileName );
+		strncpy( szFileName, pszFileName, sizeof(szFileName) );
 	}
 	else
 	{	
-- 
2.17.1

